Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/BackgroundModeStatus.kt

```diff
@@ -21,41 +21,41 @@ class BackgroundModeStatus @Inject constructor(
     @AppScope private val appScope: CoroutineScope
 ) {
 
-    val isBackgroundRestricted: Flow<Boolean?> = callbackFlow<Boolean> {
+    val isBackgroundRestricted: Flow<Boolean> = flow {
         while (true) {
             try {
-                send(pollIsBackgroundRestricted())
-            } catch (e: Exception) {
-                Timber.w(e, "isBackgroundRestricted failed.")
-                cancel("isBackgroundRestricted failed", e)
+                emit(pollIsBackgroundRestricted())
+                delay(POLLING_DELAY_MS)
+            } catch (e: CancellationException) {
+                Timber.d("isBackgroundRestricted was cancelled")
+                break
             }
-
-            if (!isActive) break
-
-            delay(POLLING_DELAY_MS)
         }
     }
         .distinctUntilChanged()
+        .onCompletion {
+            if (it != null) Timber.w(it, "isBackgroundRestricted failed.")
+        }
         .shareLatest(
             tag = "isBackgroundRestricted",
             scope = appScope
         )
 
-    val isAutoModeEnabled: Flow<Boolean> = callbackFlow<Boolean> {
+    val isAutoModeEnabled: Flow<Boolean> = flow {
         while (true) {
             try {
-                send(pollIsAutoMode())
-            } catch (e: Exception) {
-                Timber.w(e, "autoModeEnabled failed.")
-                cancel("autoModeEnabled failed", e)
+                emit(pollIsAutoMode())
+                delay(POLLING_DELAY_MS)
+            } catch (e: CancellationException) {
+                Timber.d("isAutoModeEnabled was cancelled")
+                break
             }
-
-            if (!isActive) break
-
-            delay(POLLING_DELAY_MS)
         }
     }
         .distinctUntilChanged()
+        .onCompletion {
+            if (it != null) Timber.w(it, "autoModeEnabled failed.")
```

Comment:
How can the value ever be `null`?

Reply:
If we cancel the coroutine, we just break the `while(true)` and the `flow { }` ends, then we would complete without exception.

The implicit signature here is

```kotlin
.onCompletion { it: Throwable? ->

}
```